.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,107 @@
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '41 17 * * 5'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java', 'javascript' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python' ]
+        # Learn more:
+        # https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#changing-the-languages-that-are-analyzed
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    # - name: Autobuild
+    #   uses: github/codeql-action/autobuild@v1
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    # java build
+    - name: Set up JDK 1.8
+      if: ${{ matrix.language == 'java' }} 
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Cache Maven packages
+      if: ${{ matrix.language == 'java' }} 
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2
+        key: maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          maven-
+    - name: Build and test with Maven
+      if: ${{ matrix.language == 'java' }} 
+      run: mvn --batch-mode package
+
+    # javascript build
+    - name: Setup Node.js 12.x
+      uses: actions/setup-node@v2.2.0
+      if: ${{ matrix.language == 'javascript' }}
+      with:
+        node-version: 12.x
+    - name: Cache NPM packages
+      if: ${{ matrix.language == 'javascript' }}
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-config-editor-ui-${{ hashFiles('config-editor/config-editor-ui/package.json') }}
```

Comment:
would you like to cache package.json  or package-lock.json?

Reply:
I think `package.json` since same stuff we are doing in `build-js-config-editor-ui` phase
```suggestion
      - name: Cache NPM packages
        uses: actions/cache@v2
        with:
          path: ~/.npm
          key: npm-config-editor-ui-${{ hashFiles('config-editor/config-editor-ui/package.json') }}
          restore-keys: |
            npm-config-editor-ui-
```